enginetest/evaluation.go

```diff
@@ -443,6 +444,12 @@ func runQueryPreparedWithCtx(
 	return rows, sch, err
 }
 
+// DoltCommitType is a special equality type when we expect to see a commit hash in the results
+type DoltCommitType string
+
+var DoltCommit DoltCommitType = "DOLT_COMMIT"
+var hashRegex = regexp.MustCompile(`^[0-9a-v]{32}$`)
```

Comment:
What if GMS declared an interface, something like `CustomTestValueValidator`, and it provided a `validate(value interface{}) bool` method? Then engine tests could use that interface and seems like we could push the Dolt-specific code back into the Dolt package. 

Kinda annoying to jump through that hoop for another layer of abstraction, but I think it's worth it to keep GMS 100% free of any Dolt-specific references.